src/frontend/src/lib/components/convert/ConvertInputAmount.svelte

```diff
@@ -0,0 +1,84 @@
+<script lang="ts">
+	import { IconClose } from '@dfinity/gix-components';
+	import { debounce, isNullish, nonNullish } from '@dfinity/utils';
+	import { BigNumber } from '@ethersproject/bignumber';
+	import InputCurrency from '$lib/components/ui/InputCurrency.svelte';
+	import type { Token } from '$lib/types/token';
+	import { invalidAmount } from '$lib/utils/input.utils';
+	import { parseToken } from '$lib/utils/parse.utils';
+
+	export let token: Token;
+	export let amount: number | undefined = undefined;
+	export let disabled: boolean | undefined = undefined;
+	export let customValidate: (userAmount: BigNumber | undefined) => void = () => {};
```

Comment:
in SendInputAmount we declare it as following, therefore for consistency I would keep the same:

```
	export let customValidate: (userAmount: BigNumber) => Error | undefined = () => undefined;
```

Reply:
Thanks for the explanation, makes sense. However, not fully convinced by the API.

- Should it be use a more meaningful return type than `void`? Like `{success: boolean}`?

- Shouldn't the default value be `undefined` that way the property becomes optional, similar to the other custom validation that I shared?

Reply:
So I reworked it a bit, and now it's pretty similar to the approach we have in "send". To me, looks much cleaner now. WDYT?

Reply:
This is how it looks from the parent side:
```
if (isNullish(totalFee)) {
    return;
}
...
if (userAmount.gt(parsedSendBalance)) {
	return 'insufficient-balance';
}

if (nonNullish(parsedTotalFee) && parsedTotalFee.add(userAmount).gt(parsedSendBalance)) {
	return 'insufficient-balance-for-fee';
}
```

and 

```
<ConvertInputAmount
		slot="amount"
		token={$sourceToken}
		bind:amount={sendAmount}
		{customValidate}
		disabled={isZeroBalance}
		bind:errorType
	/>
```

Reply:
- how is `errorType` used in the parent?

- what do you think of assigning the value to `errorType` in the parent and not in the `ConvertInputAmount`, would that be cleaner?

Reply:
This is how the error vars look like in the parent component:
<img width="684" alt="Screenshot 2024-11-12 at 14 58 01" src="https://github.com/user-attachments/assets/d325c388-c6e6-4a75-96a2-ae8e582306fd">

I just tried moving `errorType` up to the parent, and to be honest the current way seems to be a bit more cleaner (mostly, because we would need to call `customValidate` when value is `undefined` to reset errors, as well as we would still need to pass somehow to `ConvertInputAmount` information about the error to display it with proper CSS classes).

Reply:
P.S.: I updated the type string to use `funds` instead of `balance` for consistency.

Reply:
Alright, thanks for trying and explanation. Let's do the following. We go on with the PR but, if you don't mind, after the first usage of this component and pattern - before we use it everywhere - you allow me to have just a quick look, just in case I suddently remember a pattern we would use in other dapps once I got the code in front of my eyes 😉.

Reply:
Agreed 🙂